lib/apiservers/engine/backends/container.go

```diff
@@ -671,13 +673,13 @@ func (c *Container) ContainerAttach(prefixOrName string, ca *backend.ContainerAt
 	vc := cache.ContainerCache().GetContainer(prefixOrName)
 
 	if vc == nil {
-		//FIXME: If we didn't find in the cache, we should goto the port layer and
-		//see if it exists there.  API server might have been bounced.  For now,
-		//just return error
-
 		return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", prefixOrName))
 	}
 
+	if running, err := IsContainerRunning(vc.ContainerID); !running {
```

Comment:
There will be.  That will be the event monitoring in the portlayer that will be forthcoming.  Clint is adding that for ps and will propogate it to inspect, and then we'll use it.

The race condition will still persist even with the event monitoring.  This is just a sanity check so we don't go try to attach to a container that hasn't started.  From the user perspective, they will get a message that it is stopped and to try and start it before attaching.  They will simply check status and retry again.  If this ends up being a field problem for us, we can always tie it more tightly to the forthcoming event subscription.